src/ChildProcess/PooledInvoker.php

```diff
@@ -0,0 +1,122 @@
+<?php
+namespace React\Filesystem\ChildProcess;
+
+use React\Filesystem\AdapterInterface;
+use React\ChildProcess\Process;
+use React\Promise\Deferred;
+
+class PooledInvoker
+{
+    /**
+     * @var LoopInterface
+     */
+    protected $loop;
+
+    /**
+     * @var int
+     */
+    protected $maxSimultaneousOperations = 64;
+
+    /**
+     * All the operations that are currently being performed
+     * @var Process[]
+     */
+    protected $operations = [];
+
+    /**
+     * The command that will be called on each invoke
+     * @var string
+     */
+    protected $command;
+
+    /**
+     * PooledInvoker Constructor
+     *
+     * @param AdapterInterface $adapter
+     * @param int $maxSimultaneousOperations
+     */
+    public function __construct(AdapterInterface $adapter, $maxSimultaneousOperations = 133)
+    {
+        $this->loop    = $adapter->getLoop();
+        $this->adapter = $adapter;
+        $this->maxSimultaneousOperations = $maxSimultaneousOperations;
+        $this->command = 'php '.__DIR__.'/Command.php';
```

Comment:
Keep [this](https://github.com/reactphp/child-process#prepending-commands-with-exec) in mind otherwise you might end up with detached child processes eating up CPU usage.